R/simulate.r

```diff
@@ -139,40 +139,27 @@ simulate_chains <- function(index_cases,
                             generation_time = NULL,
                             t0 = 0,
                             tf = Inf) {
-  # Input checking
-  checkmate::assert_count(
-    index_cases,
-    positive = TRUE
-  )
-  checkmate::assert_choice(
-    statistic,
-    choices = c("size", "length")
-  )
-  .check_offspring_func_valid(offspring_dist)
-  checkmate::assert(
-    is.infinite(stat_max) ||
-      checkmate::assert_integerish(stat_max, lower = 0)
-  )
-  checkmate::assert(
-    is.infinite(pop) ||
-      checkmate::assert_integerish(pop, lower = 1)
-  )
-  checkmate::assert_number(
-    percent_immune,
-    lower = 0, upper = 1
-  )
-  if (!missing(generation_time)) {
-    .check_generation_time_valid(generation_time)
-  } else if (!missing(tf)) {
-    stop("If `tf` is specified, `generation_time` must be specified too.")
-  }
-  checkmate::assert_numeric(
-    t0,
-    lower = 0, finite = TRUE
-  )
-  checkmate::assert_number(
-    tf,
-    lower = 0
+  # Check inputs
+  func_name <- as.character(match.call()[[1]])
```

Comment:
is there any benefit of doing this over just saying `func_name = "simulate_chains"` below which would be simpler?

Reply:
It's just a guard against potential name changes.